libgadget/slotsmanager.c

```diff
@@ -61,7 +61,7 @@ int
 slots_convert(int parent, int ptype, int placement)
 {
     /*Set old slot as garbage*/
-    if(P[parent].PI > 0 && SLOTS_ENABLED(P[parent].Type))
+    if(P[parent].PI >= 0 && SLOTS_ENABLED(P[parent].Type))
```

Comment:
Wow. How could this happen...

Reply:
I know, ouch! I guess it is just so rare it never happened in tests before. Introduced by the starform branch.